### PR TITLE
Notifications UI

### PR DESCRIPTION
**Pull request by biyeun**
- builds on https://github.com/dimagi/commcare-hq/pull/9750 (merge in first) [REVIEW AFTER "icons 4 everyone"]
- behind a feature flag (needs some backend love)

Screenshots
screen shot 2016-02-04 at 5 48 03 pm

@dimagi/product

### CODE REVIEW

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 1]
```diff
@@ -150,13 +150,15 @@ def domains_for_user(context, request, selected_domain=None):
     ctxt = {
         'domain_list': domain_list,
         'current_domain': selected_domain,
-        'DOMAIN_TYPE': context['DOMAIN_TYPE']
+        'DOMAIN_TYPE': context['DOMAIN_TYPE'],
+        'can_publish_to_exchange': (
+            selected_domain is not None and selected_domain != 'public'
+            and request.couch_user and request.couch_user.can_edit_apps() and
+                (request.couch_user.is_member_of(selected_domain)
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: E131 continuation line unaligned for hanging indent

**Comment by amsagoff** [comment]
:on: :point_up:

File: corehq/apps/notifications/models.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    'FakeNotification', ['isRead', 'content', 'url', 'date', 'type']
+)
+
+
+class NotificationType(object):
+    INFO = 'info'
+    ALERT = 'alert'
+
+
+def get_fake_notifications():
```
Comment (benrudolph) [inline corehq/apps/notifications/models.py]: :trollface:

File: corehq/apps/notifications/static/notifications/spec/NotificationsService.spec.mo.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    }
+                };
+            };
+        };
+
+    var fakePromise = new FakePromise({});
+    sinon.stub(jQuery, 'ajax', fakePromise.mock);
+
+    it('Initialization', function () {
+        viewModel = $('#js-settingsmenu-notifications').startNotificationsService(fakeRMIUrl);
```
Comment (benrudolph) [inline corehq/apps/notifications/static/notifications/spec/NotificationsService.spec.mo.js]: i don't think this actually tests anything. mocha doesn't fail the test if there's javascript error

**Comment by benrudolph** [comment]
:heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat:

**Comment by czue** [comment]
very slick!!! :snowboarder:

**Comment by biyeun** [comment]
ready to be merged! fyi...this is behind a feature flag was waiting on the parent pr to get merged

**Comment by benrudolph** [comment]
ok looks good and i'll go ahead an merge, but think we should either remove that test that i commented on or make it useful

**Pull request merged**